### PR TITLE
Minimize workflow runs on Dependabot updates

### LINKED ISSUE #12911: Dependabot upgrade of CodeQL fail on pushes
### Describe the bug

The hint of dependabot in the log: 

> Error: Workflows triggered by Dependabot on the "push" event run with read-only access. Uploading Code Scanning results requires write access. To use Code Scanning with Dependabot, please ensure you are using the "pull_request" event for this workflow and avoid triggering on the "push" event for Dependabot branches. See https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#scanning-on-push for more information on how to configure these events.

The error message in the build:

```
RequestError [HttpError]: Resource not accessible by integration
    at /home/runner/work/_actions/github/codeql-action/v2.1.12/node_modules/@octokit/request/dist-node/index.js:66:23
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Job.doExecute (/home/runner/work/_actions/github/codeql-action/v2.1.12/node_modules/bottleneck/light.js:405:18) {
  status: 403,
  headers: {
    'access-control-allow-origin': '*',
    'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
    connection: 'close',
    'content-encoding': 'gzip',
    'content-security-policy': "default-src 'none'",
    'content-type': 'application/json; charset=utf-8',
    date: 'Mon, 04 Jul 2022 19:49:19 GMT',
    'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
    server: 'GitHub.com',
    'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
    'transfer-encoding': 'chunked',
    vary: 'Accept-Encoding, Accept, X-Requested-With',
    'x-content-type-options': 'nosniff',
    'x-frame-options': 'deny',
    'x-github-media-type': 'github.v3; format=json',
    'x-github-request-id': '07C4:177F:6401D6:702DC0:62C3443F',
    'x-ratelimit-limit': '1000',
    'x-ratelimit-remaining': '971',
    'x-ratelimit-reset': '1656964986',
    'x-ratelimit-resource': 'core',
    'x-ratelimit-used': '29',
    'x-xss-protection': '0'
  },
  request: {
    method: 'PUT',
    url: 'https://api.github.com/repos/keycloak/keycloak/code-scanning/analysis/status',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'CodeQL-Action/2.1.12 octokit-core.js/3.1.2 Node.js/16.[13](https://github.com/keycloak/keycloak/runs/7185892949?check_suite_focus=true#step:5:14).0 (linux; x64)',
      authorization: 'token [REDACTED]',
      'content-type': 'application/json; charset=utf-8'
    },
    body: '{"workflow_run_id":2612026[14](https://github.com/keycloak/keycloak/runs/7185892949?check_suite_focus=true#step:5:15)8,"workflow_name":"CodeQL JS Adapter","job_name":"analyze","analysis_key":".github/workflows/codeql-js-adapter-analysis.yml:analyze","commit_oid":"70c1270a0d05db1c2310235eb403024740b5e675","ref":"refs/heads/dependabot/github_actions/aquasecurity/trivy-action-0.5.1","action_name":"init","action_ref":"v2.1.12","action_oid":"unknown","started_at":"2022-07-04T19:49:[17](https://github.com/keycloak/keycloak/runs/7185892949?check_suite_focus=true#step:5:18).396Z","action_started_at":"2022-07-04T[19](https://github.com/keycloak/keycloak/runs/7185892949?check_suite_focus=true#step:5:20):49:17.[39](https://github.com/keycloak/keycloak/runs/7185892949?check_suite_focus=true#step:5:40)6Z","status":"starting","runner_os":"Linux","action_version":"2.1.12","matrix_vars":"null","runner_arch":"X64"}',
    request: { agent: [Agent], hook: [Function: bound bound register] }
  },
  documentation_url: 'https://docs.github.com/rest'
}
```

### Version

999-SNAPSHOT

### Expected behavior

Build should succeed

### Actual behavior

Build fails, see

https://github.com/keycloak/keycloak/runs/7185892949?check_suite_focus=true

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ahus1**
Workflows will only run when their own workflow file changes, not on other workflow files. Dependabot branches are ignored for "push" builds, will only run on "pull" builds as code scanning on dependabot is not available for "push" builds.

Closes #12911

### CODE REVIEW

**Comment by ahus1** [comment]
Reran builds due to failing DNS resolution of `127.0.0.1.nip.io`

File: .github/workflows/trivy-analysis.yml [hunk 1]
```diff
@@ -3,6 +3,10 @@ on:
   workflow_dispatch:
   schedule:
     - cron: "0 6 * * *"
+  pull_request:
```
[the remark this answers is not in the document]
Comment (abstractj) [reply .github/workflows/trivy-analysis.yml:6]: @ahus1 Please remove, and I can give you some context on why. The Trivy analysis run against our container images, the frequency of the build of those images is on the daily basis for nightly images, and a week or more for released images (that's on demand). That means that if we trigger Trivy on every pull-request, we are increasing the pool of runners when it is not necessary. Added to that, thankfully we don't have a CVE on our dependencies emerging every day.
Please ping me if you have any questions.

[the remark this answers is not in the document]
Comment (ahus1) [reply .github/workflows/trivy-analysis.yml]: ok, removed it. Waiting for the build to complete.

**Review by abstractj — APPROVED** [review]
Changes LGTM, I'm just waiting for the CI checks.

**Pull request merged**